htmresearch/frameworks/nlp/model_factory.py

```diff
@@ -35,6 +41,34 @@
 """
 
 
+
+class ClassificationModelTypes(object):
+  """ Enumeration of supported classification model types, mapping userland
+  identifier to constructor.  See createModel() for actual factory method
+  implementation.
+  """
+  CioWordFingerprint = ClassificationModelFingerprint
+  CioDocumentFingerprint = ClassificationModelFingerprint
+  Keywords = ClassificationModelKeywords
+  HTMNetwork = ClassificationModelHTM
+  DocumentFingerPrint = ClassificationModelDocumentFingerprint
+  cioword = ClassificationModelFingerprint
+  docfp = ClassificationModelDocumentFingerprint
+  htm = ClassificationModelHTM
+  keywords = ClassificationModelKeywords
+
+
+  @classmethod
+  def getTypes(cls):
+    types = {attr
```

Comment:
I originally was using enum, backported from python 3.4, which implemented a __members__ attr to differentiate between class attrs, and enumerations.  I had to back off since it only allows mappings of unique values and had to add this method to make up for it.  What do you suggest to make it simpler?  It's as simple as it gets, given that it returns a set of class attrs for which the value is an acceptable model constructor -- replacing the role of __members__ in enum.

This function is used by the arg parser to print the set of acceptable values for model type, as well as in the model factory to ensure that the user passes a proper value.  I.e it provides programmatic access to acceptable values, something not currently possible with the inconsistent use of strings between different model factory implementations.